docs/dev/core/internal/upload.md

```diff
@@ -0,0 +1,69 @@
+# Upload mechanism
+
+The `glean-core` Rust crate does not handle the ping upload directly.
+Network stacks vastly differ between platforms, applications and operating systems.
+It is best to use the available platform capabilities to implement any network communication.
+
+Glean still controls all upload and coordinates the platform side with its own internals.
+All wrappers implement ping uploading around a common API.
```

Comment:
```suggestion
All language bindings implement ping uploading around a common API and protocol.
```

Reply:
Is wrapper the right word here? Do we use it elsewhere in this same context? What do you think of "language binding" or "platform-specific implementation" ?

Reply:
language binding!
I didn't feel comfortable with wrapper to begin with, but the only other word my brain came up with was platform, and that didn't fit either.